tests/50federation/30room-join.pl

```diff
@@ -953,7 +953,117 @@ sub assert_is_valid_pdu {
 
             # XXX currently synapse fails with a 500 here. I'm not really convinced that's
             # a thing we want to enforce, but we don't really have a specced way to say
-            # "a remote server did something wierd".
+            # "a remote server did something weird".
+            Future->done(1);
+         }),
+      )
+   };
+
+test "Outbound federation event from other origin",
+   requires => [ local_user_fixture(), $main::INBOUND_SERVER,
+                 federation_user_id_fixture() ],
+
+   do => sub {
+      my ( $user, $inbound_server, $creator_id ) = @_;
+
+      my $versionprefix     = "v2";
+      my $local_server_name = $inbound_server->server_name;
+      my $datastore         = $inbound_server->datastore;
+
+      my $room_alias = "#50fed-room-alias:$local_server_name";
+      my $room = $datastore->create_room(
+         creator => $creator_id,
+         alias   => $room_alias,
+      );
+
+      my $room_id = $room->room_id;
+
+      my $event = $room->create_and_insert_event(
+         sender      => "\@test:$local_server_name",
+         type        => "test",
+         room_id     => $room_id,
+         state_key   => "",
+         content     => {
+            body    => "Test",
+         },
+      );
+
+      # The event was generated with a valid signature, modify it with an
+      # invalid signature.
```

Comment:
this isn't changing the signature so much as invalidating the existing one

Reply:
Yes, I failed to update this when modifying the code.